src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteItemWriter.java

```diff
@@ -0,0 +1,216 @@
+            }
+
+            for (int index = 0; index < keyIndices.length; index++)
+            {
+                this.tagsStatement.setLong(FIRST_COLUMN, atlasId);
+                this.tagsStatement.setInt(SECOND_COLUMN, keyIndices[index]);
+                this.tagsStatement.setInt(THIRD_COLUMN, valueIndices[index]);
+                this.tagsStatement.execute();
+            }
+
+            // INSERT spatial information if enabled
+            if (this.rtreeStatementOptional.isPresent())
+            {
+                final PreparedStatement unwrappedStatement = this.rtreeStatementOptional.get();
+
+                final Rectangle bounds = item.bounds();
+                final Location lowerLeft = bounds.lowerLeft();
+                final Location upperRight = bounds.upperRight();
+                final double minLon = lowerLeft.getLongitude().asDegrees();
+                final double minLat = lowerLeft.getLatitude().asDegrees();
+                final double maxLon = upperRight.getLongitude().asDegrees();
+                final double maxLat = upperRight.getLatitude().asDegrees();
+
+                // feature id and bounding box in rtree
+                unwrappedStatement.setLong(FIRST_COLUMN, atlasId);
+                unwrappedStatement.setDouble(SECOND_COLUMN, minLon);
+                unwrappedStatement.setDouble(THIRD_COLUMN, maxLon);
+                unwrappedStatement.setDouble(FOURTH_COLUMN, minLat);
+                unwrappedStatement.setDouble(FIFTH_COLUMN, maxLat);
+
+                unwrappedStatement.execute();
+            }
+        }
+
+        this.connection.commit();
+        this.connection.setAutoCommit(true);
+    }
+
+    private Iterable<? extends AtlasItem> getItemIterableOfCorrectType(final PackedAtlas atlas,
+            final ItemType itemType)
+    {
+        final Iterable<? extends AtlasItem> items;
+        switch (itemType)
+        {
+            case NODE:
+                items = atlas.nodes();
+                break;
+            case EDGE:
+                items = atlas.edges();
+                break;
+            case POINT:
+                items = atlas.points();
+                break;
+            case LINE:
+                items = atlas.lines();
+                break;
+            case AREA:
+                items = atlas.areas();
+                break;
+            default:
+                throw new CoreException("Unknown entity type {}", itemType);
+        }
+
+        return items;
+    }
+
+    private byte[] getWKBOfItem(final AtlasItem item)
+    {
+        byte[] wkb = null;
+
+        if (item.getType() != this.itemType)
+        {
+            throw new CoreException(
+                    "ItemType {} of AtlasItem {} did not match expected ItemType {}",
+                    item.getType(), item, this.itemType);
+        }
+
+        switch (item.getType())
+        {
+            case NODE:
```

Comment:
Nit: To simplify, consider breaking this down into `LineItem` and `LocationItem` since `Edge`/`Line` processing is the identical, as is `Point`/`Node` processing.